runtime/compiler/compile/J9SymbolReferenceTable.hpp

```diff
@@ -462,6 +486,39 @@ class SymbolReferenceTable : public OMR::SymbolReferenceTableConnector
    typedef TR::typed_allocator<ResolvedFieldShadowsEntry, TR::Allocator> ResolvedFieldShadowsAlloc;
    typedef std::map<ResolvedFieldShadowKey, TR::SymbolReference*, std::less<ResolvedFieldShadowKey>, ResolvedFieldShadowsAlloc> ResolvedFieldShadows;
 
+   struct FlattenedArrayElementFieldShadowKey
```

Comment:
`ResolvedFieldShadowKey` and `ResolvedFieldShadows` are used for flattened instance fields as well.  Using the following as an example, if we use `ResolvedFieldShadowKey` (`containingClass, offset, type`) for both instance fields and array element fields, `line.p1.x` and `line.p1.y` could end up having the same key as below. And for `containerInst.line.p1.x`, it is final. For `containerArr[i].line.p1.y`, `isFinal` is `false` since array element is mutable. Wouldn't that be an issue? 

`containerArr[i].line.p1.y` has key `(ContainerClass, 4, Int32)` and `isFinal=false`.

`containerInst.line.p1.x` has key `(ContainerClass, 4, Int32)` and `isFinal=true` (assume `objectHeaderSizeInBytes` is 4)

```
primitive public class ContainerClass {
   public Line2D line;
} 
primitive public class Line2D {
   public Point2D p1;
   public Point2D p2;
}
primitive public class Point2D {
   public final int x;
   public final int y;
}

ContainerClass containerInst; 

ContainerClass[] containerArr = new ContainerClass[5]; 
```

Reply:
Yes, that's a good reason to keep `_flattenedArrayElementFieldShadows` separate from `_resolvedFieldShadows`. However, it's fine for both of those maps to have the same type

Reply:
Addressed in f4541ce